### PR TITLE
refactor message generation

### PR DESCRIPTION
**Pull request by dirk-thomas**
Connects to ros2/ros2#48.

### CODE REVIEW

**Comment by esteve** [comment]
+1

File: rosidl_typesupport_opensplice_cpp/cmake/rosidl_typesupport_opensplice_cpp_generate_interfaces.cmake [hunk 1]
```diff
@@ -27,53 +27,60 @@ rosidl_generate_dds_interfaces(
...
-  elseif("${_extension}" STREQUAL ".srv")
-    get_filename_component(name "${_idl_file}" NAME_WE)
-    list(APPEND _dds_idl_files "${_dds_idl_path}/Sample${name}Request_.idl")
-    list(APPEND _dds_idl_files "${_dds_idl_path}/Sample${name}Response_.idl")
+  get_filename_component(_name "${_idl_file}" NAME_WE)
+  if("${_extension} " STREQUAL ".msg ")
+    get_filename_component(_parent_folder "${_idl_file}" DIRECTORY)
+    get_filename_component(_parent_folder "${_parent_folder}" NAME)
+    list(APPEND _dds_idl_files
+      "${_dds_idl_base_path}/${PROJECT_NAME}/${_parent_folder}/dds_opensplice/${_name}_.idl")
```
Comment (wjwwood) [inline rosidl_typesupport_opensplice_cpp/cmake/rosidl_typesupport_opensplice_cpp_generate_interfaces.cmake]: Are/should the `msg` related dds files getting generated into a `msg` folder?

It seems like the `srv` related dds files are (see below on line 39).

**Comment by wjwwood** [comment]
Otherwise, +1

File: rosidl_typesupport_opensplice_cpp/cmake/rosidl_typesupport_opensplice_cpp_generate_interfaces.cmake (hunk 1 above)
Comment (dirk-thomas) [inline rosidl_typesupport_opensplice_cpp/cmake/rosidl_typesupport_opensplice_cpp_generate_interfaces.cmake]: That is exactly what is happening here: `_parent_folder` contains either `msg` or `srv` depending on if it is a plain message or a request / response message.

Comment (wjwwood) [inline rosidl_typesupport_opensplice_cpp/cmake/rosidl_typesupport_opensplice_cpp_generate_interfaces.cmake]: Ah, I see, I was confused at first as to why the `.srv` extension got different treatment. lgtm

**Comment by tfoote** [comment]
+1

**Pull request merged**